src/Plugin/dgi_migrate_alter/foxml/DgisNodesAlter.php

```diff
@@ -0,0 +1,287 @@
+    $process['field_record_information'][3]['values']['_field_record_creation_date_start'][0]['query'] =
+      'mods:recordInfo/mods:recordCreationDate[@point="start"] or mods:originInfo/mods:dateCreated[@point="start"]';
+    $process['field_record_information'][3]['values']['_field_record_creation_date_end'][0]['query'] =
+      'mods:recordInfo/mods:recordCreationDate[@point="end"] or mods:originInfo/mods:dateCreated[@point="end"]';
+    $process['field_record_information'][3]['values']['field_record_information_note'][] = [
+      'plugin' => 'single_value',
+    ];
+    $process['field_record_information'][3]['values']['field_record_information_note'][] = [
+      'plugin' => 'callback',
+      'callable' => 'array_filter',
+    ];
+    $process['field_record_information'][3]['values']['field_record_information_note'][] = [
+      'plugin' => 'null_coalesce',
+    ];
+    $process['field_related_item_paragraph'][3]['values']['field_related_item_genre'][] = [
+      'plugin' => 'single_value',
+    ];
+    $process['field_related_item_paragraph'][3]['values']['field_related_item_genre'][] = [
+      'plugin' => 'callback',
+      'callable' => 'array_filter',
+    ];
+    $process['field_related_item_paragraph'][3]['values']['field_related_item_genre'][] = [
+      'plugin' => 'null_coalesce',
+    ];
+
+    $process['_rights_statement'] = $process['_resource_type'];
+    $process['_rights_statement'][0]['query'] = 'mods:accessCondition[@type="rights statement"]';
+    $process['_rights_statement'][4]['values']['_vid'][0]['default_value'] = 'rights_statements';
+
+    $process['_unspecified_rights_statement'] = $process['_unspecified_resource_type'];
+    $process['_unspecified_rights_statement'][1]['use_as_key'] = '@_rights_statement';
+    $process['_unspecified_rights_statement'][2]['values']['_vid'][0]['default_value'] = 'rights_statements';
+
+    $process['field_rights_statement'] = $process['field_resource_type'];
+    $process['field_rights_statement'][0]['source'] = [
+      '@_rights_statement',
+      '@_unspecified_rights_statement',
+    ];
+
+    $process['field_publication_genre'][3] = [
+      'plugin' => 'single_value',
+    ];
+
+    array_splice($process['field_publication_genre'], 4, 0, [
+      [
+        'plugin' => 'callback',
+        'callable' => 'array_filter',
+      ],
+      [
+        'plugin' => 'null_coalesce',
+      ],
+    ]);
+
+    $process['field_publication_title'][3] = [
+      'plugin' => 'single_value',
+    ];
+
+    array_splice($process['field_publication_title'], 4, 0, [
+      [
+        'plugin' => 'callback',
+        'callable' => 'array_filter',
+      ],
+      [
+        'plugin' => 'null_coalesce',
+      ],
+    ]);
+
+    $process['field_remote_media_url'] = $process['field_ismn'];
+    $process['field_remote_media_url'][0]['query'] = 'mods:identifier[@displayLabel="remote media URL"]';
+
```

Comment:
For both `field_extent_first_page` and `field_extent_last_page` the `@unit` attribute is missing on the XPath do we care?

```
'field_extent_first_page' => array (7) [
        0 => array (5) [
            'plugin' => string (37) "dgi_migrate.process.xml.context_query"
            'missing_behavior' => string (12) "skip_process"
            'source' => string (11) "@_mods_node"
            'xpath' => string (12) "@_mods_xpath"
            'query' => string (32) "mods:part/mods:extent/mods:start"
        ]
```
```
    'field_extent_last_page' => array (7) [
        0 => array (5) [
            'plugin' => string (37) "dgi_migrate.process.xml.context_query"
            'missing_behavior' => string (12) "skip_process"
            'source' => string (11) "@_mods_node"
            'xpath' => string (12) "@_mods_xpath"
            'query' => string (30) "mods:part/mods:extent/mods:end"
        ]
```

`mods:relatedItem/mods:part/mods:extent[@unit="pages"]/mods:start` and `mods:relatedItem/mods:part/mods:extent[@unit="pages"]/mods:end`

Reply:
Fixed.